src/compiler/checker.ts

```diff
@@ -16743,6 +16746,65 @@ namespace ts {
             }
         }
 
+        function markDecoratorMedataDataTypeNodeAsReferenced(node: TypeNode): void {

```

Comment:
Could you add comment on why we need this separate from using `markTypeNodeAsReferenced`